pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -362,6 +365,14 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun) err
 		return controller.NewPermanentError(err)
 	}
 
+	// Ensure that the context variables in the PipelineRun parameters are valid.
+	if err := resources.ValidateContextVariables(pipelineSpec, pr); err != nil {
```

Comment:
Any reason for doing it in the reconciler loop ? we didn't do it before :monocle_face:

Reply:
thanks for the review @vdemeester!

yes, we run it in the need to do it in the reconciler look to access the pipelineruns to get the name, uid etc -- @sbwsg and I explored trying to do it in other ways and this was the only option we had

yes, we didn't have validation for context variables before

Reply:
Ah, I think I may have just realised a mistake I made - I suggested doing this in the reconciler when we were intending to implement annotations-as-variables.  When annotations are used we have to do the validation in the reconciler because there's no way to validate the annotations referenced by a Pipeline without also having the PipelineRun object - e.g. a Pipeline can use `$(context.pipelineRun.annotations.foo-bar-baz)` and we can't validate `foo-bar-baz` is correct until we also have the PipelineRun available to us.

However we then decided to push the annotations-as-variables into a separate PR because we figured there'd be more design and discussion required.  So, now that we're only validating the use of uid and names I think we probably can push the validation into the webhook, since the set of available variables is known to us in advance.  What do you think @jerop?

Reply:
Right, that is, indeed my initial thought on this. If we have annotations and/or labels in the variables, then there needs to be a "reconciler" validation, but for the rest (uid, …), it can be done solely on the webhook side :upside_down_face:

Reply:
yes yes, the issue was with annotations...just updated the validation